src/clinical/service-worker-thread/tasks.ts

```diff
@@ -71,27 +78,67 @@ const sortDocs = (sort: string) => (currentRecord: ClinicalInfo, nextRecord: Cli
   return order;
 };
 
-const mapEntityDocuments = (totalDonors: number, schemas: any, query: ClinicalQuery) => ([
-  entityName,
-  results,
-]: [string, ClinicalInfo[]]) => {
-  if (isEmpty(results)) return undefined;
+const mapEntityDocuments = (
+  totalDonors: number,
+  schemas: any,
+  query: ClinicalQuery,
+  completionStats: CompletionRecord[],
+) => (
+  entity: [string, ClinicalInfo[]],
+  index: number,
+  originalResultsArray: [string, ClinicalInfo[]][],
+) => {
```

Comment:
Since this method is a bit complicated it would benefit from some comments explaining the work that is done in it and what the objective is. The return type of the inner method would be nice to declare as well.

Reply:
I ended up refactoring this a fair bit with comments and it should be much more readable